timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.corruption.detection;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.paxos.Client;
+import com.palantir.paxos.ImmutableNamespaceAndUseCase;
+import com.palantir.paxos.NamespaceAndUseCase;
+import com.palantir.paxos.PaxosAcceptorState;
+import com.palantir.paxos.PaxosStateLog;
+import com.palantir.paxos.PaxosValue;
+import com.palantir.paxos.SqliteConnections;
+import com.palantir.paxos.SqlitePaxosStateLog;
+import com.palantir.timelock.Constants;
+import com.palantir.timelock.history.LocalHistoryLoader;
+import com.palantir.timelock.history.PaxosLogHistoryProvider;
+import com.palantir.timelock.history.TimeLockPaxosHistoryProvider;
+import com.palantir.timelock.history.models.AcceptorUseCase;
+import com.palantir.timelock.history.models.LearnerUseCase;
+import com.palantir.timelock.history.remote.TimeLockPaxosHistoryProviderResource;
+import com.palantir.timelock.history.sqlite.SqlitePaxosStateLogHistory;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+import org.immutables.value.Value;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public final class TimeLockCorruptionTestSetup implements TestRule {
+    private TemporaryFolder tempFolder = new TemporaryFolder();
+    private DataSource localDataSource;
+    private DataSource remoteDataSource1;
+    private DataSource remoteDataSource2;
+    private PaxosLogHistoryProvider paxosLogHistoryProvider;
+
+    private StateLogComponents defaultLocalServer;
+    private List<StateLogComponents> defaultRemoteServerList;
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return RuleChain.outerRule(tempFolder)
+                .around(new ExternalResource() {
+                    @Override
+                    protected void before() {
+                        try {
+                            setup();
+                        } catch (Throwable throwable) {
+                            // no op
```

Comment:
Interesting - if we get an exception on startup, we don't want to do anything? I'd have thought we'd do:
```java
throw new RuntimeException(throwable, "failed on startup");
```